hydromt/data_adapter_v1/geodataframe_adapter.py

```diff
@@ -0,0 +1,313 @@
+from hydromt.nodata import NoDataStrategy, _exec_nodata_strat
+from hydromt.typing import GEOM_TYPES, ErrorHandleMethod, TotalBounds
+
+from .data_adapter_base import DataAdapterBase
+
+logger: Logger = getLogger(__name__)
+
+
+class GeoDataFrameAdapter(DataAdapterBase):
+    """The GeoDataFrameAdapter performs transformations on GeoDataFrames."""
+
+    source: GeoDataFrameDataSource
+    used: bool = False
+
+    def get_data(
+        self,
+        bbox: list[float] | None = None,
+        mask: gpd.GeoDataFrame | None = None,
+        buffer: float = 0.0,
+        variables: list[str] | None = None,
+        predicate: str = "intersects",
+        handle_nodata: NoDataStrategy = NoDataStrategy.RAISE,
+        logger: Logger | None = None,
+    ) -> gpd.GeoDataFrame:
+        """Read in data and transform them to HydroMT standards."""
+        gdf: gpd.GeoDataFrame = self.source.read_data(
+            bbox, mask, buffer, variables, predicate, handle_nodata, logger=logger
+        )
+        self.used = True  # mark used
+        # rename variables and parse crs & nodata
+        gdf = self._rename_vars(gdf)
+        gdf = self._set_crs(gdf, logger=logger)
+        gdf = self._set_nodata(gdf)
+        # slice
+        gdf = GeoDataFrameAdapter._slice_data(
+            gdf, variables, mask, bbox, buffer, predicate, handle_nodata, logger=logger
+        )
+        # uniformize
+        gdf = self._apply_unit_conversions(gdf, logger=logger)
+        gdf = self._set_metadata(gdf)
+        return gdf
+
+    def _rename_vars(self, gdf: gpd.GeoDataFrame):
+        # rename and select columns
+        if rename := self.source.rename:
+            rename = {k: v for k, v in rename.items() if k in gdf.columns}
+            gdf = gdf.rename(columns=rename)
+        return gdf
+
+    def _set_crs(self, gdf: gpd.GeoDataFrame, logger=logger):
+        if crs := self.source.crs is not None and gdf.crs is None:
+            gdf.set_crs(crs, inplace=True)
+        elif gdf.crs is None:
+            raise ValueError(
+                f"GeoDataFrame {self.source.name}: CRS not defined in data catalog or data."
+            )
+        elif self.crs is not None and gdf.crs != pyproj.CRS.from_user_input(self.crs):
+            logger.warning(
+                f"GeoDataFrame {self.source.name}: CRS from data catalog does not match CRS of"
+                " data. The original CRS will be used. Please check your data catalog."
+            )
+        return gdf
+
+    def _slice_data(
+        gdf: gpd.GeoDataFrame,
+        variables: str | list[str] | None = None,
+        geom: GEOM_TYPES | None = None,
+        bbox: GEOM_TYPES | None = None,
+        crs: CRS | None = None,
+        buffer: float = 0.0,
+        predicate: str = "intersects",  # TODO: enum available predicates
+        handle_nodata: NoDataStrategy = NoDataStrategy.RAISE,  # TODO: review NoDataStrategy + axes
+        logger: Logger = logger,
+    ) -> gpd.GeoDataFrame:
```

Comment:
In the latest no data PR we landed on data returning functions can optionally turn None if the nodata strategy was set to IGNORE and triggred, since that's easier to check for

Reply:
It is difficult to "coerce" nodata in the driver (it just does not read anything), as wel as the metadata resolver (it does not find anything), so perhaps we always raise there and catch an error in the adapter and return None? I do not want to check for `None` so often.

Reply:
While I agree with you, we'll have to check for errors regularly anyway, whether that's by checking for None, or using a try/except I don't really care that much. @hboisgon seemed to have a preference for None though